distributed/diagnostics/tests/test_worker_plugin.py

```diff
@@ -136,6 +136,33 @@ async def test_release_dep_called(c, s, w):
     await async_wait_for(lambda: not (w.task_state or w.dep_state), timeout=10)
 
 
+@pytest.mark.asyncio
+async def test_registering_with_name_arg():
+    class FooWorkerPlugin:
+        def setup(self, worker):
+            if hasattr(worker, "foo"):
+                raise RuntimeError(f"Worker {worker.address} already has foo!")
+
+            worker.foo = True
+
+    async with LocalCluster(
```

Comment:
Rather than use LocalCluster I recommend using `gen_cluster`, and then following up with another worker

```python
@gen_cluster(client=True)
async def test_registering_with_name_arg(c, s, a, b):
    ...
    async with Worker(s.address) as w:
        ...

```

See https://distributed.dask.org/en/latest/develop.html#writing-tests for more information

Reply:
Thank you. I didn't know how to add a worker after `gen_cluster`.